zkllvm/use-cases/primer.mdx

```diff
@@ -0,0 +1,222 @@
+# Primer
+
+This primer acts as an intro guide to writing complex circuits and compiling them. 
+
+## Dependency management
+
+Most circuits that cover common use cases (e.g., circuits verifying state transitions) require the use of external dependencies. 
+
+In such cases, [**using `clang` or `rustc` directly**](../getting-started/compiling-a-circuit) is discouraged. Instead, it is preferable to use a dedicated build management system:
+
+* CMake for C++
+* Cargo for Rust
+
+:::tip[`clang`]
+
+When calling `clang` directly, dependency management can also be handled via the following methods:
+
+* Via the CLI
+* Via env variables
+* Via updating the system path
+
+Using CMake is still preferable as it offers several valuable features (such as external library detection) that simplify working with complex circuits.
+
+:::
+
+:::info
+
+This primer uses `crypto3` (C++) and `arkworks` (Rust) to show how to use external libraries in a circuit. The primer can be reused for any other suitable library.
+
+:::
+
+### Using CMake
+
+Switch to the working directory:
+
+```bash
+mkdir new_circuit && cd new_circuit
+```
+
+Create a directory for CMake modules:
+
+```bash
+mkdir cmake && cd cmake
+```
+
+Add the [**`CircuitCompile.cmake` module from the zkLLVM repository**](https://github.com/NilFoundation/zkLLVM/blob/master/cmake/CircuitCompile.cmake) to the `./new_circuit/cmake` directory:
```

Comment:
SInce we require `zkllvm` package to be installed, `CircuitCompile.cmake` is a part of it, so the copying is not necessary, we can simply adjust module path to its fixed location:
```cmake
list(APPEND CMAKE_MODULE_PATH "/usr/lib/zkllvm/share/zkllvm")
```

Reply:
I guess `find_package(zkllvm)` works here, isn't it? So user won't need to hardcode the path

Reply:
No, it does not work.

Reply:
Hmm, that's weird, how am I supposed to ship cmake modules then through apt packages...

Reply:
@makxenov I see, this makes more sense! I've revised the flow and the CMake config to mention this